File: cpp/repl/big-numbers2/token.cpp
#include "token.h"

Token::Token() {

}

Token::~Token() {

}

std::string Token::getLiteral() {
    return tokenLiteral;
}

void Token::setLiteral(std::string tokLiteral) {
    tokenLiteral = tokLiteral;
}

TokenType Token::getType() {
    return type;
}

void Token::setType(TokenType tokenType) {
    type = tokenType;
}

std::string Token::toString() {
    return "Token [" + token_type_converter[type] + "] <" + tokenLiteral + ">";
}

File: java/VmSimulation/src/token/Token.java
package token;

public class Token {
    public TokenType type;
    public String literal;

    public Token () {

    }

    public Token(TokenType type, String literal) {
        this.type = type;
        this.literal = literal;
    }

    public TokenType getType() {
        return type;
    }

    @Override
    public String toString() {
        return "Token [" + type + "] <" + literal + ">";
    }
}
